### ISSUE TITLE
morpheus provider failes on portal deleted items

### ISSUE THREAD

**Issue by mbjorgan** [gomorpheus/terraform-provider-morpheus#164]
Scenario:
You have defined a task (or any other resource) in terraform code.
You apply the change -> task is created in morpheus portal (we then have it in the code, the statefile and the portal)
All good so far.
You then delete the task in the portal and run terraform apply again (now we have the task in the statefile and in the code)
This is what happens then:
```hcl
Plan: 1 to add, 1 to change, 0 to destroy.
╷
│ Error: Job not found with id 34
│ 
│   with morpheus_task_job.Update_contacts,
│   on morpheus_task_job.tf line 88, in resource "morpheus_task_job" "Update_contacts":
│   88: resource "morpheus_task_job" "Update_contacts" {
│ 
╵

Error: Process completed with exit code 1.
```
Terraform should be able to figure out that the task is missing and then re-add the missing task.

**Comment by martezr** [comment]
The provider code needs to be updated to properly handle read operations that don't find the defined resource and update the resource ID to an empty string to force Terraform to create the missing resource.

**Comment by olliephillips** [comment]
This looks like would be a straightforward thing to implement, it is only the number or resources which would make it an item of work.

I have just tested this which would need to be implemented in the read function of each resource. This forces the recreation of the resource.

```
if err != nil {
		if resp != nil && resp.StatusCode == 404 {
			log.Printf("API 404: %s - %s", resp, err)
			log.Printf("Forcing recreation of resource")
			d.SetId("")
			return diags
			//return diag.FromErr(err)
		} else {
			log.Printf("API FAILURE: %s - %s", resp, err)
			return diag.FromErr(err)
		}
}
```

Would you recommend an alternative approach @martezr?

**Comment by olliephillips** [comment]
If this is the way to go I'm happy to pick this up.

**Comment by martezr** [comment]
@olliephillips That's the recommended approach, I updated the code for a couple of the newest resources and the task job resource.

**Comment by olliephillips** [comment]
Hi @mbjorgan, the merged PR #184 should fix this for majority of resources. Please review in next release and close issue if good for you 👍

**Comment by mbjorgan** [comment]
Awesome @olliephillips 👍 
I'll upgrade the provider when there's a new release and test the changes :)

**Comment by martezr** [comment]
PR #184 has been included in the v0.9.7 release of the provider that was just released.
